test/pkg/firewall/firewall.go

```diff
@@ -1,126 +1,335 @@
-		return fmt.Errorf("failed to check nftables.conf on debug pod: %w", err)
+		return fmt.Errorf("failed to marshal to JSON: %w", err)
 	}
 
-	// Convert output to an integer and check if the include statement exists
-	includeCount, err := strconv.Atoi(strings.TrimSpace(string(output)))
+	fileName := fmt.Sprintf("convertedData-%s.json", role)
+	filePath := filepath.Join(artifactsDir, fileName)
+	err = utilsHelpers.WriteFile(filePath, jsonData)
 	if err != nil {
-		return fmt.Errorf("failed to parse grep output: %w", err)
+		return err
 	}
 
-	if includeCount > 0 {
-		log.Println("Include statement already exists, no need to add it")
-		return nil
+	obj := &machineconfigurationv1.MachineConfig{}
+	if err := json.Unmarshal(jsonData, obj); err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %w", err)
 	}
 
-	addCommand := `echo 'include "/etc/nftables/firewall.nft"' >> /host/etc/sysconfig/nftables.conf`
-	_, err = RunRootCommandOnPod(debugPod, addCommand, false, utilsHelpers)
+	// Check if the MachineConfig already exists
+	existingMC := &machineconfigurationv1.MachineConfig{}
+	err = c.Get(context.TODO(), types.NamespacedName{Name: obj.Name}, existingMC)
 	if err != nil {
-		return fmt.Errorf("failed to edit nftables.conf on debug pod: %w", err)
+		if errors.IsNotFound(err) {
+			// If it doesn't exist, create it
+			if err := c.Create(context.TODO(), obj); err != nil {
+				return fmt.Errorf("failed to create MachineConfig: %w", err)
+			}
+			fmt.Println("MachineConfig created successfully.")
+		} else {
+			return fmt.Errorf("failed to get MachineConfig: %w", err)
+		}
+	} else {
+		// If it exists, update it
+		// Set the resourceVersion from the existing object
+		obj.ResourceVersion = existingMC.ResourceVersion
+
+		if err := c.Update(context.TODO(), obj); err != nil {
+			return fmt.Errorf("failed to update MachineConfig: %w", err)
+		}
+		fmt.Println("MachineConfig updated successfully.")
 	}
 
 	return nil
 }
 
-func NftListAndWriteToFile(debugPod *v1.Pod, utilsHelpers utils.UtilsInterface, artifactsDir, fileName string) ([]byte, error) {
-	command := "nft list ruleset"
-	output, err := RunRootCommandOnPod(debugPod, command, true, utilsHelpers)
-	if err != nil {
-		return nil, fmt.Errorf("failed to list NFT ruleset on node %s: %w", debugPod.Spec.NodeName, err)
+// Helper function to convert map[interface{}]interface{} to map[string]interface{}.
+func convertMapInterfaceToString(data interface{}) interface{} {
+	switch v := data.(type) {
+	case map[interface{}]interface{}:
+		newMap := make(map[string]interface{})
+		for key, value := range v {
+			newMap[fmt.Sprintf("%v", key)] = convertMapInterfaceToString(value)
+		}
+		return newMap
+	case []interface{}:
+		for i, item := range v {
+			v[i] = convertMapInterfaceToString(item)
+		}
 	}
+	return data
+}
 
-	if len(output) == 0 {
-		return nil, fmt.Errorf("no nft rules on node %s: ", debugPod.Spec.NodeName)
-	}
+func UpdateMachineConfiguration(c *client.ClientSet) error {
```

Comment:
I don't think this func belongs to firewall package

Reply:
From my understanding this is not updating a MC, it's creating a new one.
Can you just use a MC object and apply it with the client? 
You already added the scheme here https://github.com/openshift-kni/commatrix/pull/55/files#diff-2895f30116602d8fe6b0545c186d4f81247fa4eaab049dab174c6c91c0036e08R49 no need to create an additional client
Also maybe use a more meaningful name for the MC than "cluster"